### PR TITLE
Update 2021-05-27-Testing-Theory.markdown

### PR DESCRIPTION
**Pull request by NikUrs**
- Комментарии к статье "in my humble opinion")

- Завёл некоторые баг-репорты и импрувменты к сайту "ilarionhalushka.github.io".

### БАГ-РЕПОРТЫ

**Баг репорт 01.** Слово "автору" накладывается на слово "Выразить" в названии кнопки "Выразить благодарность автору" в мобильной версии сайта.
https://docs.google.com/spreadsheets/u/1/d/13eJVcux9o0er66fm639u94eIWB7ryvMg_ocCam6CVhw/edit?usp=drive_web&ouid=102064553302234595178

**Баг репорт 02.** Пользователь не перенаправляется к описанию элемента списка после нажатия на "Продвинутые техники тест-дизайна" внутри блока "Навигация".
https://docs.google.com/spreadsheets/d/1rfAnroE2EuT8zhjtmxhT7kis5kfd1iU9dmVeo-nzyUE/edit#gid=0

**Баг репорт 03.** Иконка социальной сети "Facebook" не отображается в кнопке перехода на эту сеть.
https://docs.google.com/spreadsheets/d/1WJfnkpuc0k-_f3FnPqvruS_nbGOhpbtQJMEFM2Pqqbs/edit#gid=0

**Баг репорт 04.** Кнопки блока социальных сетей в шапке сайта не имеют состояния "hover" и "active".
https://docs.google.com/spreadsheets/d/1uOXxgl3oOmT0KpKW4XFYH5DMo9hQHA76PXfS3OfwArg/edit#gid=0

**Баг репорт 05.** Текст кнопки "Теорія тестування українською" сливается с фоном кнопки.
https://docs.google.com/spreadsheets/d/1ajhMKG-efvGsmBKaLQo-bTIrbA258nj4s1jP6noIpR4/edit#gid=0

**Баг репорт 06.** Картинка "Testing types" не увеличивается после клика на неё.
https://docs.google.com/spreadsheets/d/19zBsDYu8EyLKt8KbhrhHIbOLwlOZkZXwzeEWXX19NSs/edit#gid=0

**Баг репорт 07.** Виды функционального и нефункционального тестирования, отображённые в тексте статьи и на схеме "Testing types" не соответствуют друг другу.
https://docs.google.com/spreadsheets/d/11WYo1wzzVeSn0PWaelJJk_VSrRqxccwvrMWb0AQjcSs/edit#gid=0

**Баг репорт 08.** Ответ от сервера содержит статус код 404 после перехода по гиперссылке "time" в статье "Soft Skills of Senior Developer".
https://docs.google.com/spreadsheets/d/1BcFaTqnMq64o3ol659HPfCVaUq1R-5sXrgvl8NPr0_g/edit#gid=0

**Баг репорт 09.** Код статуса ответа от сервера отображается как 404 после перехода по второй гиперссылке "Теорія тестування українською 🇺🇦" из статьи "Теория тестирования от А до Я".
https://docs.google.com/spreadsheets/d/1jLwy48bzdTk5yyxXGoqF8RZB0xrvaIXA_nBvHLo_oAc/edit#gid=0

### ИМПРУВМЕНТ-РЕПОРТЫ

**Импрувмент-репорт 01.** Текстовый курсор в блоке "IT Blog by Ilarion Halushka" не отображён с классической анимацией "появление - исчезновение".
https://docs.google.com/spreadsheets/d/1meeA4eLDlrvXyTQUmsVTSgXwGKC_EpaXH5ZwLcABQ_E/edit?usp=drive_link

**Импрувмент репорт 02**. Картинка статьи "How To Make Developers Perform" содержит текст на русском языке.
https://docs.google.com/spreadsheets/d/11EySoM51-t07MOAqk7apU0JqLpOS7dk9828OYYnY0U8/edit#gid=0

**Импрувмент-репорт 03.** Каждая из статей не содержит своё уникальное превью в блоке "ALSO ON ILARIONHALUSHKA".
https://docs.google.com/spreadsheets/d/1_8qdGnE66yCnCxo2te4X4vY82RotWdYL0WLy9FK5EjM/edit#gid=0

### CODE REVIEW

File: _posts/2021-05-27-Testing-Theory.markdown [hunk 1]
```diff
@@ -598,6 +609,12 @@ UI-дефекты например. Но аппаратный сбой, ника
 Обнаружение и исправление дефектов не помогут, 
 если созданная система не подходит пользователю и не удовлетворяет его ожиданиям и потребностям.
 
+<
+Сколько бы мы не находили ошибок, это не даст нам гарантию того, что мы нашли их все или что продукт будет качественным.
+Даже если наш продукт соответствует всем требованиям, если все тесты выполнены, и ошибки не обнаружены, это ещё не гарантия того, что ПО будет удовлетворять  потребностям и ожиданиям пользователей. 
+https://vladislaveremeev.gitbook.io/qa_bible/obshee/principy-testirovaniya
+https://crashtest.by/7-principles-of-testing/
```
Comment (IlarionHalushka) [inline _posts/2021-05-27-Testing-Theory.markdown]: please add under "Источники" section:
https://vladislaveremeev.gitbook.io
https://crashtest.by

File: _posts/2021-05-27-Testing-Theory.markdown [hunk 2]
```diff
@@ -598,6 +609,12 @@ UI-дефекты например. Но аппаратный сбой, ника
 Обнаружение и исправление дефектов не помогут, 
 если созданная система не подходит пользователю и не удовлетворяет его ожиданиям и потребностям.
 
+<
+Сколько бы мы не находили ошибок, это не даст нам гарантию того, что мы нашли их все или что продукт будет качественным.
+Даже если наш продукт соответствует всем требованиям, если все тесты выполнены, и ошибки не обнаружены, это ещё не гарантия того, что ПО будет удовлетворять  потребностям и ожиданиям пользователей.
```
Comment (IlarionHalushka) [inline _posts/2021-05-27-Testing-Theory.markdown:613]: 👍

File: _posts/2021-05-27-Testing-Theory.markdown [hunk 3]
```diff
@@ -200,7 +211,7 @@ image_title="test design error guessing" %}
 
 Звучит сложно, но на практике использовать эту технику очень просто и логично.
 
-- Суть техники — мы не проверяем все сочетания всех значений, но проверяем ВСЕ ПАРЫ значений.
+- Суть техники — мы не проверяем все сочетания всех значений, но проверяем ВСЕ <УНИКАЛЬНЫЕ> ПАРЫ значений.
```
Comment (IlarionHalushka) [inline _posts/2021-05-27-Testing-Theory.markdown]:
```suggestion
- Суть техники — мы не проверяем все сочетания всех значений, но проверяем ВСЕ УНИКАЛЬНЫЕ ПАРЫ значений.
```

File: _posts/2021-05-27-Testing-Theory.markdown [hunk 4]
```diff
@@ -154,6 +159,12 @@ image_title="test design equvivalence" %}
 
 ### Анализ Граничных Значений (Boundary Value Analysis) <a target="_blank" rel="noreferrer" href="https://youtu.be/7a7vGZqwtCA?t=1661" title="Youtube"><i class="fa fa-1x fa-youtube-square"></i></a>
 
+<
+Техника проверки поведения продукта на крайних (граничных) значениях входных данных. 
+Суть техники состоит в том, что для тестирования требуется взять границу, где один класс эквивалентности переходит в другой и протестировать границу, до и после неё. Берётся максимально приближённое значение слева от границы и максимально приближённое значение справа от границы. При этом учитывается шаг изменения значений входных данных. 
+https://qaevolution.ru/testovaya-dokumentaciya/test-dizajn/texnika-analiza-granichnyx-znachenij/
```
Comment (IlarionHalushka) [inline _posts/2021-05-27-Testing-Theory.markdown]: please add the link under "Источники" section:

qaevolution.ru

File: _posts/2021-05-27-Testing-Theory.markdown [hunk 5]
```diff
@@ -154,6 +159,12 @@ image_title="test design equvivalence" %}
 
 ### Анализ Граничных Значений (Boundary Value Analysis) <a target="_blank" rel="noreferrer" href="https://youtu.be/7a7vGZqwtCA?t=1661" title="Youtube"><i class="fa fa-1x fa-youtube-square"></i></a>
 
+<
+Техника проверки поведения продукта на крайних (граничных) значениях входных данных. 
+Суть техники состоит в том, что для тестирования требуется взять границу, где один класс эквивалентности переходит в другой и протестировать границу, до и после неё. Берётся максимально приближённое значение слева от границы и максимально приближённое значение справа от границы. При этом учитывается шаг изменения значений входных данных.
```
Comment (IlarionHalushka) [inline _posts/2021-05-27-Testing-Theory.markdown:164]: like, но не совсем понял 

```
При этом учитывается шаг изменения значений входных данных. 
```

File: _posts/2021-05-27-Testing-Theory.markdown [hunk 6]
```diff
@@ -140,6 +140,11 @@ tags: best, testing, qa, interview
 
 ### Эквивалентное Разделение (Equivalence Partitioning) <a target="_blank" rel="noreferrer" href="https://youtu.be/7a7vGZqwtCA?t=1352" title="Youtube"><i class="fa fa-1x fa-youtube-square"></i></a>
 
+<
```
Comment (IlarionHalushka) [inline _posts/2021-05-27-Testing-Theory.markdown]: pls remove <

File: _posts/2021-05-27-Testing-Theory.markdown [hunk 7]
```diff
@@ -140,6 +140,11 @@ tags: best, testing, qa, interview
...
+Техника, при которой мы разделяем функционал (часто диапазон возможных вводимых значений) на группы эквивалентных по своему влиянию на систему значений. Такое разделение помогает убедиться в правильном функционировании целой системы — одного класса эквивалентности, проверив только один элемент этой группы. Эта техника заключается в разбиении всего набора тестов на классы эквивалентности с последующим сокращением числа тестов.
+https://qaevolution.ru/testovaya-dokumentaciya/test-dizajn/texnika-analiza-klassov-ekvivalentnosti/
+>
```
Comment (IlarionHalushka) [inline _posts/2021-05-27-Testing-Theory.markdown]: pls remove >

File: _posts/2021-05-27-Testing-Theory.markdown [hunk 8]
```diff
@@ -154,6 +159,12 @@ image_title="test design equvivalence" %}
 
 ### Анализ Граничных Значений (Boundary Value Analysis) <a target="_blank" rel="noreferrer" href="https://youtu.be/7a7vGZqwtCA?t=1661" title="Youtube"><i class="fa fa-1x fa-youtube-square"></i></a>
 
+<
```
Comment (IlarionHalushka) [inline _posts/2021-05-27-Testing-Theory.markdown]: pls remove <

File: _posts/2021-05-27-Testing-Theory.markdown [hunk 9]
```diff
@@ -154,6 +159,12 @@ image_title="test design equvivalence" %}
...
+Техника проверки поведения продукта на крайних (граничных) значениях входных данных. 
+Суть техники состоит в том, что для тестирования требуется взять границу, где один класс эквивалентности переходит в другой и протестировать границу, до и после неё. Берётся максимально приближённое значение слева от границы и максимально приближённое значение справа от границы. При этом учитывается шаг изменения значений входных данных. 
+https://qaevolution.ru/testovaya-dokumentaciya/test-dizajn/texnika-analiza-granichnyx-znachenij/
+>
```
Comment (IlarionHalushka) [inline _posts/2021-05-27-Testing-Theory.markdown]: pls remove >

File: _posts/2021-05-27-Testing-Theory.markdown [hunk 10]
```diff
@@ -598,6 +609,12 @@ UI-дефекты например. Но аппаратный сбой, ника
 Обнаружение и исправление дефектов не помогут, 
 если созданная система не подходит пользователю и не удовлетворяет его ожиданиям и потребностям.
 
+<
```
Comment (IlarionHalushka) [inline _posts/2021-05-27-Testing-Theory.markdown]: pls remove <

File: _posts/2021-05-27-Testing-Theory.markdown [hunk 11]
```diff
@@ -598,6 +609,12 @@ UI-дефекты например. Но аппаратный сбой, ника
...
+Сколько бы мы не находили ошибок, это не даст нам гарантию того, что мы нашли их все или что продукт будет качественным.
+Даже если наш продукт соответствует всем требованиям, если все тесты выполнены, и ошибки не обнаружены, это ещё не гарантия того, что ПО будет удовлетворять  потребностям и ожиданиям пользователей. 
+https://vladislaveremeev.gitbook.io/qa_bible/obshee/principy-testirovaniya
+https://crashtest.by/7-principles-of-testing/
+>
```
Comment (IlarionHalushka) [inline _posts/2021-05-27-Testing-Theory.markdown]: pls remove >

File: _posts/2021-05-27-Testing-Theory.markdown [hunk 12]
```diff
@@ -140,6 +140,11 @@ tags: best, testing, qa, interview
 
 ### Эквивалентное Разделение (Equivalence Partitioning) <a target="_blank" rel="noreferrer" href="https://youtu.be/7a7vGZqwtCA?t=1352" title="Youtube"><i class="fa fa-1x fa-youtube-square"></i></a>
 
+<
+Техника, при которой мы разделяем функционал (часто диапазон возможных вводимых значений) на группы эквивалентных по своему влиянию на систему значений. Такое разделение помогает убедиться в правильном функционировании целой системы — одного класса эквивалентности, проверив только один элемент этой группы. Эта техника заключается в разбиении всего набора тестов на классы эквивалентности с последующим сокращением числа тестов.
```
Comment (IlarionHalushka) [inline _posts/2021-05-27-Testing-Theory.markdown]: Формулировка "Такое разделение помогает убедиться в правильном функционировании целой системы" не совсем корректна, т.к. нам ничего не дает гарантию в `правильном функционировании целой системы`...

**Review by IlarionHalushka — CHANGES_REQUESTED** [review]
Thank you!!

**Comment by IlarionHalushka** [comment]
Nice, thank you again. Pushed fixes for the most of the bugs from your list: https://github.com/IlarionHalushka/ilarionhalushka.github.io

The most important thing is that I could understand all the bugs just by reading the title. 🚀 

Some of the bugs from your ist are actually improvements but still all are make sense and are valuable. Thank you again!

File: _posts/2021-05-27-Testing-Theory.markdown (hunk 6 above)
Comment (NikUrs) [reply _posts/2021-05-27-Testing-Theory.markdown]: done

File: _posts/2021-05-27-Testing-Theory.markdown (hunk 12 above)
Comment (NikUrs) [reply _posts/2021-05-27-Testing-Theory.markdown]: У каждого ресурса, смотрю, своё объяснение, сложно подобрать или сформулировать простое и понятное определение, чтоб была только самая суть. В общем, может, подойдёт, добавил несколько других предложений (взял с www.learnqa.ru).

File: _posts/2021-05-27-Testing-Theory.markdown (hunk 7 above)
Comment (NikUrs) [reply _posts/2021-05-27-Testing-Theory.markdown]: done

File: _posts/2021-05-27-Testing-Theory.markdown (hunk 8 above)
Comment (NikUrs) [reply _posts/2021-05-27-Testing-Theory.markdown]: done

File: _posts/2021-05-27-Testing-Theory.markdown (hunk 5 above)
Comment (NikUrs) [reply _posts/2021-05-27-Testing-Theory.markdown:164]: Согласен, как-то сумбурно получилось)
Тут хотел указать, что важно обращать внимание именно на то, с каким шагом наши тестовые данные изменяются. Если шаг 0.1 и тестируем границу 1.0, то мы берём 0.9 и 1.1; если шаг 0.01 и тестируем границу 1.0, то берём 0.99 и 1.01.
Когда нужно протестировать, что платёжный терминал принимает 1000 гривен (как условную границу), а минимальная купюра, которую он принимает 100грн., то для тестирования границы мы берём 900грн. и 1100грн.

File: _posts/2021-05-27-Testing-Theory.markdown (hunk 4 above)
Comment (NikUrs) [reply _posts/2021-05-27-Testing-Theory.markdown]: done

File: _posts/2021-05-27-Testing-Theory.markdown (hunk 9 above)
Comment (NikUrs) [reply _posts/2021-05-27-Testing-Theory.markdown]: done

File: _posts/2021-05-27-Testing-Theory.markdown (hunk 3 above)
Comment (NikUrs) [reply _posts/2021-05-27-Testing-Theory.markdown]: done

File: _posts/2021-05-27-Testing-Theory.markdown (hunk 10 above)
Comment (NikUrs) [reply _posts/2021-05-27-Testing-Theory.markdown]: done

File: _posts/2021-05-27-Testing-Theory.markdown (hunk 1 above)
Comment (NikUrs) [reply _posts/2021-05-27-Testing-Theory.markdown]: done

File: _posts/2021-05-27-Testing-Theory.markdown (hunk 11 above)
Comment (NikUrs) [reply _posts/2021-05-27-Testing-Theory.markdown]: done

**Comment by NikUrs** [comment]
> Nice, thank you again. Pushed fixes for the most of the bugs from your list: https://github.com/IlarionHalushka/ilarionhalushka.github.io
> 
> The most important thing is that I could understand all the bugs just by reading the title. 🚀
> 
> Some of the bugs from your ist are actually improvements but still all are make sense and are valuable. Thank you again!

Thank you a lot for feedback and for your high-quality learning materials!
